File: spendzero/mobile/lib/design_system/typography.dart
import 'package:flutter/widgets.dart';

import 'colors.dart';

/// Typography primitives matching Lovable's `font-display` (Fraunces) /
/// `font-sans` (Inter) split and the exact px sizes used in `index.tsx`
/// and sibling routes. Tailwind's arbitrary `text-[Npx]` values map 1:1 to
/// `fontSize: N`. Use these factories instead of `Theme.of(context).textTheme`
/// so every screen draws from the same exact-pixel source.
class DSType {
  DSType._();

  static const String display = 'Fraunces';
  static const String sans = 'Inter';

  static TextStyle display_(
    double size, {
    FontWeight weight = FontWeight.w600,
    double? height,
    double letterSpacing = -0.4,
    Color color = DSColors.foreground,
    FontStyle style = FontStyle.normal,
  }) {
    return TextStyle(
      fontFamily: display,
      fontSize: size,
      fontWeight: weight,
      height: height ?? 1.05,
      letterSpacing: letterSpacing,
      color: color,
      fontStyle: style,
    );
  }

  static TextStyle sans_(
    double size, {
    FontWeight weight = FontWeight.w400,
    double? height,
    double letterSpacing = 0,
    Color color = DSColors.foreground,
  }) {
    return TextStyle(
      fontFamily: sans,
      fontSize: size,
      fontWeight: weight,
      height: height,
      letterSpacing: letterSpacing,
      color: color,
    );
  }

  /// `text-[11px] uppercase tracking-[0.28em]` — eyebrow/overline labels.
  static TextStyle eyebrow({
    double size = 11,
    double trackingEm = 0.28,
    Color color = DSColors.mutedForeground,
    FontWeight weight = FontWeight.w500,
  }) {
    return sans_(size, weight: weight, letterSpacing: size * trackingEm, color: color);
  }
}

File: spendzero/mobile-rn/design-system/typography.ts
import { TextStyle } from "react-native";
import { colors } from "./colors";

/**
 * Typography primitives matching Lovable's `font-display` (Fraunces) /
 * `font-sans` (Inter) split and the exact px sizes used in `index.tsx`.
 * Tailwind's arbitrary `text-[Npx]` values map 1:1 to `fontSize: N`.
 */
export const fontFamily = {
  display: "Fraunces-Variable",
  sans: "Inter-Variable",
} as const;

export function display(
  size: number,
  opts: {
    weight?: TextStyle["fontWeight"];
    height?: number;
    letterSpacing?: number;
    color?: string;
    italic?: boolean;
  } = {},
): TextStyle {
  const { weight = "600", height, letterSpacing = -0.4, color = colors.foreground, italic = false } = opts;
  return {
    fontFamily: fontFamily.display,
    fontSize: size,
    fontWeight: weight,
    lineHeight: height ?? size * 1.05,
    letterSpacing,
    color,
    fontStyle: italic ? "italic" : "normal",
  };
}

export function sans(
  size: number,
  opts: {
    weight?: TextStyle["fontWeight"];
    height?: number;
    letterSpacing?: number;
    color?: string;
  } = {},
): TextStyle {
  const { weight = "400", height, letterSpacing = 0, color = colors.foreground } = opts;
  return {
    fontFamily: fontFamily.sans,
    fontSize: size,
    fontWeight: weight,
    lineHeight: height,
    letterSpacing,
    color,
  };
}

/** `text-[11px] uppercase tracking-[0.28em]` — eyebrow/overline labels. */
export function eyebrow(opts: {
  size?: number;
  trackingEm?: number;
  color?: string;
  weight?: TextStyle["fontWeight"];
} = {}): TextStyle {
  const { size = 11, trackingEm = 0.28, color = colors.mutedForeground, weight = "500" } = opts;
  return sans(size, { weight, letterSpacing: size * trackingEm, color });
}
